lib/puppet/http/client.rb

```diff
@@ -59,6 +62,11 @@ def get(url, headers: {}, params: {}, user: nil, password: nil, ssl_context: nil
 
     request = Net::HTTP::Get.new(url, @default_headers.merge(headers))
 
+    user = options.fetch(:user, nil)
+    password = options.fetch(:password, nil)
+    ssl_context = options.fetch(:ssl_context, nil)
+    include_system_store = options.fetch(:include_system_store, false)
+
```

Comment:
Since these are repeated for each method, perhaps just pass `options` to `execute_streaming` and have it fetch them?